src/state/session/index.tsx

```diff
@@ -266,7 +250,6 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
       // We never reuse agents so let's fully neutralize the previous one.
       // This ensures it won't try to consume any refresh tokens.
       prevAgent.sessionManager.session = undefined
-      setPersistSessionHandler(undefined)
```

Comment:
Note I am no longer cleaning up the persistence handler when the current agent gets swapped. I think this should be fine. It didn't used to be cleaned up in the past either, I only added this at some intermediate stage for peace of mind. But we already have a condition [here](https://github.com/bluesky-social/social-app/blob/99d1a881f2f5c16dddfc10550b39e379690c8135/src/state/session/reducer.ts#L72-L75) that prevents inactive agents from clearing persisted data. For inactive agents, `refreshedAccount` [should be](https://github.com/bluesky-social/social-app/blob/99d1a881f2f5c16dddfc10550b39e379690c8135/src/state/session/index.tsx#L65) `undefined` because we already set `prevAgent.sessionManager.session = undefined` [here](https://github.com/bluesky-social/social-app/blob/99d1a881f2f5c16dddfc10550b39e379690c8135/src/state/session/index.tsx#L268). And even if somehow it had a session object, the reducer always [matches events with state by the did](https://github.com/bluesky-social/social-app/blob/99d1a881f2f5c16dddfc10550b39e379690c8135/src/state/session/reducer.ts#L99) so it won't update the wrong thing.